mabwiser/mab.py

```diff
@@ -596,20 +595,74 @@ def __init__(self,
 
             if isinstance(neighborhood_policy, NeighborhoodPolicy.Clusters):
                 self._imp = _Clusters(self._rng, self.arms, self.n_jobs, self.backend, lp,
-                                      self.neighborhood_policy.n_clusters, self.neighborhood_policy.is_minibatch)
+                                      neighborhood_policy.n_clusters, neighborhood_policy.is_minibatch)
             elif isinstance(neighborhood_policy, NeighborhoodPolicy.Radius):
                 self._imp = _Radius(self._rng, self.arms, self.n_jobs, self.backend, lp,
-                                    self.neighborhood_policy.radius, self.neighborhood_policy.metric,
-                                    self.neighborhood_policy.no_nhood_prob_of_arm)
+                                    neighborhood_policy.radius, neighborhood_policy.metric,
+                                    neighborhood_policy.no_nhood_prob_of_arm)
             elif isinstance(neighborhood_policy, NeighborhoodPolicy.KNearest):
                 self._imp = _KNearest(self._rng, self.arms, self.n_jobs, self.backend, lp,
-                                      self.neighborhood_policy.k, self.neighborhood_policy.metric)
+                                      neighborhood_policy.k, neighborhood_policy.metric)
             else:
                 check_true(False, ValueError("Undefined context policy " + str(neighborhood_policy)))
         else:
             self.is_contextual = isinstance(learning_policy, LearningPolicy.LinUCB)
             self._imp = lp
 
+    @property
```

Comment:
MAB doesn't support custom learning policies, only the simulator does and the simulator doesn't access these properties because a class that inherits from BaseMAB won't have them.